src/test/java/cloud/eppo/helpers/TestUtils.java

```diff
@@ -0,0 +1,62 @@
+package cloud.eppo.helpers;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.*;
+
+import cloud.eppo.BaseEppoClient;
+import cloud.eppo.EppoHttpClient;
+import cloud.eppo.EppoHttpClientRequestCallback;
+import java.lang.reflect.Field;
+import okhttp3.*;
+
+public class TestUtils {
+
+  @SuppressWarnings("SameParameterValue")
+  public static void mockHttpResponse(String host, String responseBody) {
+    // Create a mock instance of EppoHttpClient
+    EppoHttpClient mockHttpClient = mock(EppoHttpClient.class);
+
+    // Mock sync get
+    Response dummyResponse =
+        new Response.Builder()
+            // Used by test
+            .code(200)
+            .body(ResponseBody.create(responseBody, MediaType.get("application/json")))
+            // Below properties are required to build the Response (but unused)
+            .request(new Request.Builder().url(host).build())
+            .protocol(Protocol.HTTP_1_1)
+            .message("OK")
+            .build();
+    when(mockHttpClient.get(anyString())).thenReturn(dummyResponse);
+
+    // Mock async get
+    doAnswer(
+            invocation -> {
+              EppoHttpClientRequestCallback callback = invocation.getArgument(1);
+              callback.onSuccess(responseBody);
+              return null; // doAnswer doesn't require a return value
+            })
+        .when(mockHttpClient)
+        .get(anyString(), any(EppoHttpClientRequestCallback.class));
+
+    setBaseClientHttpClientOverrideField(mockHttpClient);
+  }
```

Comment:
I'd honestly prefer if we used OkHttp's excellent `MockWebServer` (https://github.com/square/okhttp/tree/master/mockwebserver) but this is low priority